junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionUtils.java

```diff
@@ -147,9 +145,7 @@ static boolean objectsAreEqual(Object obj1, Object obj2) {
 		if (obj1 == null) {
 			return (obj2 == null);
 		}
-		else {
-			return obj1.equals(obj2);
-		}
+		return obj1.equals(obj2);
```

Comment:
@marcphilipp Okay, I think I was right, because when I replace the contents of `AssertionUtils.objectsAreEqual()` with `return Objects.equals(obj1, obj2); `, it causes a few tests to fail.

So, to answer your question: no, it seems that apparently the code here _isn't_ equivalent to `Objects.equals()`. :)